docs/tutorials/02_create_your_first_app/05_storing-data.md

```diff
@@ -0,0 +1,234 @@
+---
+sidebar_position: 5
+title: Step 5 - Storing data
+---
+
+# Step 5 - Storing data
+
+In this step, we will solve a problem: our actor needs to keep a list and track of *how many votes* each programming language (e.g. "Rust") has.
+
+## Creating a data structure to store the data
+
+In order to store the potential choices and how many votes each has, we need a collection. We will use a Motoko data structure called [`RBTree`](/motoko/main/base/RBTree.md) which is similar to "maps" or "dictionaries" in other languages. Essentially a key-value store.
+
+`RBTree` will "map" an *entry* of type `Text` (e.g. `"Rust"`) to the a *vote count* of type `Nat` (e.g. `5`).
+
+For example, the data structure for our "favorite programming language" poll may look like this:
+
+| id  | Vote Count |
+| --- | ------ |
+| "Motoko"   | 0 |
+| "Rust"   |   0 |
+| "TypeScript"   |  0 |
+| "Python"   | 0     |
+
+
+### Necessary Imports
+
+We need to import a few things in order to store and query the data:
+
+* To use `RBTree` we need to import `RBTree`. 
+* We need to import the standard type [`Nat`](/motoko/main/base/Nat.md) in order to use some of `RBTree`'s functions.
+* We also import `Iter` because we will use it later in this tutorial.
+
+So we add an import statement to the beginning of our file `main.mo` (and ***outside of the actor***):
+
+```motoko
+import RBTree "mo:base/RBTree";
+import Nat "mo:base/Nat";
+import Iter "mo:base/Iter";
+```
+
+### Creating an instance of the data
+
+Next we need to declare the variable of this type ***inside the actor***:
+```motoko
```

Comment:
Linebreak here - doubt this renders wrong, but a random edit could lead to it rendering wrong.

Reply:
Good point. I’ll fix

Reply:
done